test_driver/transitions_perf_test.dart

```diff
@@ -0,0 +1,231 @@
+
+      // We may want to return to the previous category later.
+      // Reset its scroll (matters for desktop layout).
+      if (demoItem != null) await scrollToTop(demoItem, driver);
+
+      // Scroll to the category list.
+      if (demoCategory != 'study') {
+        print('scrolling to $currentDemoCategory category');
+        await driver.scrollUntilVisible(
+          homeList,
+          demoList,
+          dyScroll: -1000,
+          timeout: const Duration(seconds: 10),
+        );
+      }
+    }
+
+    // Scroll to demo and open it twice.
+    demoItem = find.byValueKey(demo);
+
+    print('scrolling to demo');
+    await driver.scrollUntilVisible(
+      demoList,
+      demoItem,
+      dxScroll: -500,
+      dyScroll: -50,
+      alignment: 0.5,
+      timeout: const Duration(seconds: 10),
+    );
+
+    for (var i = 0; i < 2; i += 1) {
+      print('tapping demo');
+      await driver.tap(demoItem); // Launch the demo
+
+      sleep(const Duration(milliseconds: 500));
+
+      if (_unsynchronizedDemos.contains(demo)) {
+        await driver.runUnsynchronized<void>(() async {
+          await driver.tap(backButton);
+        });
+      } else {
+        await driver.tap(backButton);
+      }
+    }
+    print('< Success');
+  }
+
+  await scrollToTop(demoItem, driver);
+}
+
+void main([List<String> args = const <String>[]]) {
+  group('flutter gallery transitions', () {
+    FlutterDriver driver;
+    IsolatesWorkaround workaround;
+
+    setUpAll(() async {
+      driver = await FlutterDriver.connect();
+
+      // See _handleMessages() in transitions_perf.dart.
+      _allDemos = List<String>.from(json.decode(
+        await driver.requestData('demoDescriptions'),
+      ) as List<dynamic>);
+      if (_allDemos.isEmpty) throw 'no demo names found';
+
+      // TODO: Remove workaround for https://github.com/flutter/flutter/issues/24703
+      final isWeb = await driver.requestData('isWeb') == 'true';
+      if (!isWeb) {
+        workaround = IsolatesWorkaround(driver);
+        await workaround.resumeIsolates();
+      }
+
+      if (args.contains('--with_semantics')) {
+        print('Enabeling semantics...');
+        await driver.setSemantics(true);
+      }
+
+      await isPresent(galleryHeader, driver);
+    });
+
+    tearDownAll(() async {
+      if (driver != null) {
+        await driver.close();
+        await workaround.tearDown();
+      }
+    });
+
+    test('all demos', () async {
+      // Collect timeline data for just a limited set of demos to avoid OOMs.
+      final timeline = await driver.traceAction(
+        () async {
+          await runDemos(_profiledDemos, driver);
+        },
+        streams: const <TimelineStream>[
+          TimelineStream.dart,
+          TimelineStream.embedder,
+        ],
+      );
+
+      final summary = TimelineSummary.summarize(timeline);
+      await summary.writeSummaryToFile('transitions', pretty: true);
```

Comment:
How are we going to consume these summaries after the tests complete?

Reply:
That is yet to be determined